simulate/simulate.go

```diff
@@ -0,0 +1,116 @@
+package simulate
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	tp "github.com/kubearmor/KubeArmor/KubeArmor/types"
+	"sigs.k8s.io/yaml"
+)
+
+// Options Structure
+type Options struct {
+	Config string
+	Action string
+	Policy string
+}
+type SimulationOutput struct {
+	ClusterName   string
+	HostName      string
+	PodName       string
+	ContainerID   string
+	ContainerName string
+	Labels        string
+	Policy        string
+	Severity      int
+	Type          string
+	Source        string
+	Operation     string
+	Resource      string
+	Data          string
+	Action        string
+	Result        string
+}
+
+type KubeArmorCfg struct {
+	DefaultPosture string
+}
+
+func StartSimulation(o Options) error {
+	policyFile, err := os.ReadFile(filepath.Clean(o.Policy))
+	karmorPolicy := &tp.K8sKubeArmorPolicy{}
+	response := &SimulationOutput{
+		ClusterName: "Unkown",
+		HostName:    "Unkown",
+		PodName:     "Unknown",
+		Labels:      "Unkown",
+		Type:        "MatchedPolicy",
+		ContainerID: "Unkown",
+	}
+	if err != nil {
+		return fmt.Errorf("unable to read policy file; %s", err.Error())
+	}
+	js, err := yaml.YAMLToJSON(policyFile)
+	if err != nil {
+		return err
+	}
+	err = json.Unmarshal(js, karmorPolicy)
+	if err != nil {
+		return err
+	}
+	if len(karmorPolicy.Spec.Process.MatchPaths) > 0 {
+		response.Resource = karmorPolicy.Spec.Process.MatchPaths[0].Path
+		response.Policy = o.Policy
+		response.Severity = 1
+		response.Source = karmorPolicy.Spec.Process.MatchPaths[0].Path
+		response.Data = "syscall=SYS_EXECVE"
+		response.Action = karmorPolicy.Spec.Process.Action
+		response.Result = "Permission Denied"
+		PrintResults(response, "Block")
+		return nil
+	} else if len(karmorPolicy.Spec.File.MatchPaths) > 0 {
+		response.Resource = karmorPolicy.Spec.File.MatchPaths[0].Path
+		response.Policy = o.Policy
+		response.Severity = 1
+		response.Source = karmorPolicy.Spec.File.MatchPaths[0].Path
+		response.Data = "syscall=SYS_FOPEN"
```

Comment:
hey, @nyrahul  is this the right syscall?

Reply:
Nope, you can setup kubearmor and check the telemetry events for file open.